core/domain/topic_services.py

```diff
@@ -294,11 +329,39 @@ def apply_change_list(topic_id, change_list):
                 elif (change.property_name ==
                       topic_domain.TOPIC_PROPERTY_LANGUAGE_CODE):
                     topic.update_language_code(change.new_value)
+                else:
+                    raise Exception('Invalid change dict.')
+            elif (change.cmd ==
+                  subtopic_page_domain.CMD_UPDATE_SUBTOPIC_PAGE_PROPERTY):
+                subtopic_page_id = (
+                    subtopic_page_domain.SubtopicPage.get_subtopic_page_id(
+                        topic_id, change.id))
+                index = next(
```

Comment:
Two thoughts on this:
- it's a bit weird to use `index` as both the main variable and a dummy variable inside the comprehension.
- how about representing modified_subtopic_pages as a dict keyed by subtopic ID? Easier to fetch stuff that way and will simplify a lot of this. Also the representation feels more natural as there's no inherent order to those domain objects.

Reply:
Yeah, I had thought about the dict, but then thought lists might be more readable. Will change that.

Reply:
Done.